app/js/lib/foxbox/services.js

```diff
@@ -203,6 +176,35 @@ export default class Services extends EventDispatcher {
   }
 
   /**
+   * Returns in-memory service cache. When cache is not initialized we fill it
+   * in with the content of local IndexedDB.
+   *
+   * @return {Promise<Map<string, BaseService>>}
+   * @private
+   */
+  [p.getCache]() {
+    if (this[p.cache]) {
+      return this[p.cache];
+    }
+
+    return this[p.cache] = this[p.db].getServices()
```

Comment:
Here was a bug, ```[p.cache]``` should have been a promise with ```Map``` value, not just ```Map``` in this case ```getCache``` can be called simultaneously from several places and all calls will wait on the single promise. Before that it could happen that view does two requests, first creates empty map and waits for the result to populate it with, but second request thinks that map is ready (but it's still empty).